### ISSUE TITLE
VPN / Proxy issure

### ISSUE THREAD

**Issue by FeelsBadMan1** [AmmeySaini/Edu-Mail-Generator#83]
Hello, i noticed, that if i visit any collage site using firefox it give me a error telling me i cant access the site using anonymizer extentions, vpns, proxies etc. but i have none of those enabled. it also print a issure into the console. i added some screenshots, that show exactly what i mean:
![grafik](https://user-images.githubusercontent.com/40109798/125683227-2e13d764-ef84-475c-ac10-fac2aac7363b.png)
![grafik](https://user-images.githubusercontent.com/40109798/125683323-3ca4b198-565a-4990-ae8a-902da5eb9460.png)

**Comment by TheYoungMark** [comment]
Exact Same Issue Is happening to me. Chrome and firefox give me a proxy error even though Im not using a proxy or VPN

**Comment by AmmeySaini** [comment]
Try the same with undetected chrome (v0.2.7)

**Comment by TheYoungMark** [comment]
> Try the same with undetected chrome (v0.2.7)

I have tried with all the modes. It’s not just your bot though. A few other bots don’t work because of this same issue. The website was down yesterday for  maintenance so they might have changed something.

**Comment by CereProductions** [comment]
Same issue here, any way of passing through there?

**Comment by AmmeySaini** [comment]
I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.

**Comment by TheYoungMark** [comment]
> I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
> I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.

can confirm that it does work on an incognito window

**Comment by CereProductions** [comment]
> > I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
> > I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.
> 
> can confirm that it does work on an incognito window

How do you open it on an incognito window from the console?

**Comment by TheYoungMark** [comment]
> > > I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
> > > I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.
> > 
> > 
> > can confirm that it does work on an incognito window
> 
> How do you open it on an incognito window from the console?

do everything as usual and have the proxy page come up. then on that page do CTRL + SHIFT + N and paste the link of the proxy error page and itll give you a captcha and you can go trough

**Comment by CereProductions** [comment]
> > > > I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
> > > > I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.
> > > 
> > > 
> > > can confirm that it does work on an incognito window
> > 
> > 
> > How do you open it on an incognito window from the console?
> 
> do everything as usual and have the proxy page come up. then on that page do CTRL + SHIFT + N and paste the link of the proxy error page and itll give you a captcha and you can go trough

Didnt work for me, same error page

**Comment by TheYoungMark** [comment]
> > > > > I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
> > > > > I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.
> > > > 
> > > > 
> > > > can confirm that it does work on an incognito window
> > > 
> > > 
> > > How do you open it on an incognito window from the console?
> > 
> > 
> > do everything as usual and have the proxy page come up. then on that page do CTRL + SHIFT + N and paste the link of the proxy error page and itll give you a captcha and you can go trough
> 
> Didnt work for me, same error page

try with undetected. I just tested on my own pc and a server and they both work with an incognito tab. Now all we gotta do is wait for the fix

**Comment by TheYoungMark** [comment]
> > > > > I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
> > > > > I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.
> > > > 
> > > > 
> > > > can confirm that it does work on an incognito window
> > > 
> > > 
> > > How do you open it on an incognito window from the console?
> > 
> > 
> > do everything as usual and have the proxy page come up. then on that page do CTRL + SHIFT + N and paste the link of the proxy error page and itll give you a captcha and you can go trough
> 
> Didnt work for me, same error page

sorry i meant control + shift + N

**Comment by TheYoungMark** [comment]
> I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
> I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.

i tested them with all modes and tried to add the incognito mode myself. I do not know how to code but with my testing, incognito mode only works with chrome undetected

**Comment by lstprjct** [comment]
the incognito mode only works with chrome undetected

> I just checked and figured that the problem is mainly with the driver but it works well in the incongito window of the driver.
> I need someone to confirm if it's working fine in their webdrivers incongito window and I'll push an update asap if it works.

 the incognito mode only works with chrome undetected

**Comment by lstprjct** [comment]
@AmmeySaini Please release an update.

**Comment by Mtkchaudhary** [comment]
100% working with chrome undetected without icognito

**Comment by CereProductions** [comment]
> 100% working with chrome undetected without icognito

Not for me, only incognito mode but the bot doesnt vcontinue working in incognito cause it is a different window

**Comment by red0888** [comment]
Hello dev, thank you for making this script. I hope it gets fixed soon, keep up the good job! There's another repo which has over 30 colleges, you might want to have a look at it it's based on your same generator
https://github.com/Dreamsparkplus/Edu-Email-Creator-BOT

**Comment by Thenafi** [comment]
> 100% working with chrome undetected without icognito

not working  mine

**Comment by mayuroks** [comment]
@Thenafi Thanks for the comment. If you could record you screen and share a video of the working flow, that would be great. 
Tell us the exact steps to try.

**Comment by FeelsBadMan1** [comment]
Hello again, i read all these answers and tried all of them. Nothing worked. I even tried to enable incognite on chrome_undetected but it where the same issure. I then thought, maybe i enable it too slow so i implemented a option that force selenium on browserstatup to directly open it in the incognito mode. After doing so i sadly noticed, that the proxy / vpn issure still persists even tho i dont have any of those enabled. I tried using incognito on all supported browsers and i added even more browsers like brave or opera but the issure where the same.

**Comment by mayuroks** [comment]
@FeelsBadMan1 I think we should try the steps manually once. I think the website has somehow enabled bot/script detection and is showing us the error page. I think the error message is a just a generic message to show "Something is wrong" and not necessary the actual error message.

**Comment by Ricothefish** [comment]
> Hello again, i read all these answers and tried all of them. Nothing worked. I even tried to enable incognite on chrome_undetected but it where the same issure. I then thought, maybe i enable it too slow so i implemented a option that force selenium on browserstatup to directly open it in the incognito mode. After doing so i sadly noticed, that the proxy / vpn issure still persists even tho i dont have any of those enabled. I tried using incognito on all supported browsers and i added even more browsers like brave or opera but the issure where the same.

Same here

**Comment by blakemueller1243** [comment]
sadly i got it to automatically open chrome in incognito mode and that doesent work. oddly enough though if i use ctrl-shift-n and open an incognito browser and copy paste the url it works.

**Comment by Thenafi** [comment]
> @Thenafi Thanks for the comment. If you could record your screen and share a video of the working flow, that would be great.
> or you could share the exact steps to try.

https://cutt.ly/MQyDIw9  here is a video ... i think everyone having same problem

**Comment by blakemueller1243** [comment]
maybe this can be a fix? if you add proxy support i was able to get to the account creation area just fine with a proxy via a spoofer so proxies are NOT the issue

**Comment by red0888** [comment]
the issue probably lies within the chromedriver, im sure they are detecting when using automatic windows

**Comment by blakemueller1243** [comment]
thats probably exactly it sadly. i tried another customized web driver but nada :(

**Comment by red0888** [comment]
> thats probably exactly it sadly. i tried another customized web driver but nada :(

you think there's a way this can be fixed?

**Comment by AmmeySaini** [comment]
The CCC has added a new bot detection script named "Incapsula" which is detecting the webdrivers and flagging it as some anonymous source of surfing.
The news is that it can be fixed but can take time (Looking for some contributors)

**Comment by blakemueller1243** [comment]
i found these maybe these will help a bit?
https://github.com/ziplokk1/incapsula-cracker-py3
https://github.com/Imbuedhush/Incapsula-Bypass

**Comment by AmmeySaini** [comment]
@blakemueller1243 Thanks, something useful

**Comment by AmmeySaini** [comment]
Good news:
Incapsula bypassed💥
Will push the code soon.

**Comment by red0888** [comment]
Yes!

**Comment by red0888** [comment]
> Hello dev, thank you for making this script. I hope it gets fixed soon, keep up the good job! There's another repo which has over 30 colleges, you might want to have a look at it it's based on your same generator
> https://github.com/Dreamsparkplus/Edu-Email-Creator-BOT

Please consider forking this repo before the fix, they got lot of colleges!! Thanks again

**Comment by AmmeySaini** [comment]
Yes, surely if it'll not be a long work to do 😛

**Comment by AmmeySaini** [comment]
@red0888 just check that forked repo and seems like it's not coded well it fails at supplemental questions steps and you need to manually enter all those questions.

**Comment by red0888** [comment]
> @red0888 just check that forked repo and seems like it's not coded well it fails at supplemental questions steps and you need to manually enter all those questions.

I see. Perhaps you can integrate it some other day. I'm looking forward to the incapsula fix, thanks a ton!
- cheers, Red.

**Comment by blakemueller1243** [comment]
lets goooooooooo

**Comment by AmmeySaini** [comment]
Fixed!! Ref - [here](https://github.com/AmmeySaini/Edu-Mail-Generator/commit/f5ce5034107de95cfb9354dbec3f32277d15086b)

**Comment by FeelsBadMan1** [comment]
thank u for the fix. however it seems like there is a import that is invalid or something:
![grafik](https://user-images.githubusercontent.com/40109798/127904592-92d801d9-edd5-4688-b7d3-d97387a20ddf.png)

**Comment by FeelsBadMan1** [comment]
i now changed the import to: import undetected_chromedriver as uc
at the beginning it seems like it is working but then i get this error:
![grafik](https://user-images.githubusercontent.com/40109798/127905142-4d7645a5-270b-4469-97b6-65710ed87e18.png)

**Comment by AmmeySaini** [comment]
You need to rerun setup.py for the working of new version

**Comment by FeelsBadMan1** [comment]
> 
> 
> You need to rerun setup.py for the working of new version

I have done this already

**Comment by FeelsBadMan1** [comment]
I think they just banned my Ip or fingerprint idk. using a amazon aws rdp now and it seems to be working currently. however some warnings are being printed into the console when running the script. they not cause any error the script works fine but they are annoying a bit:
![grafik](https://user-images.githubusercontent.com/40109798/127910853-156a2461-67d9-4f53-a9b0-d88e44edd9ac.png)

**Comment by AmmeySaini** [comment]
Those are expected, will fix them some other day

**Comment by FeelsBadMan1** [comment]
oh okay. however thank u for the fix <3

**Comment by FuckingToasters** [comment]
I'm not at home yet so i can't try it (maybe someone here can try it before i'm at home and let me know if it avoids the proxy error)

https://github.com/nitinkumar30/edu-mail-auto-generator

Note that i have not tried this repo and the repo from the link above so far. If it do work, i'm glad i helped you.

Regards,
FuckingToasters

**Comment by FuckingToasters** [comment]
I think i found a solution to this Problem:

I've used the script and got a action required error without any captcha to solve. 
I then visited the site in a real browser and got the same error but with a captcha. 
I then solved the captcha and got the proxy error again even tho i don't use one. 
I then installed foxyproxy browser extension and added a static residential proxy. 
After adding the static residential proxy in foxyproxy that the issue where gone in he normal browser. 

As result of the steps above, i think to make it work with this script, a few edits are required:
- adding proxy support
- change element values (cause design etc. have changed so new element xpath etc.)
